### PR TITLE
feat: add basic e2e for custom ca trust

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Adds a simple e2e test that checks whether custom ca certs were properly passed to correct location during node bootstrapping process
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/scenario/scenario_ubuntu2204-custom-ca-trust.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+					},
+				}
+			},
+			VMConfigMutator: func(vmss *armcompute.VirtualMachineScaleSet) {
+				vmss.Properties.VirtualMachineProfile.StorageProfile.ImageReference = &armcompute.ImageReference{
+					ID: to.Ptr(DefaultImageVersionIDs["ubuntu2204"]),
+				}
+			},
+			LiveVMValidators: []*LiveVMValidator{
+				FileExistsInDirectoryValidator("/usr/local/share/ca-certificates/certs"),
```
Comment (cameronmeissner) [inline e2e/scenario/scenario_ubuntu2204-custom-ca-trust.go:33]: why can't we use DirectoryValidator instead?

Comment (UtheMan) [reply e2e/scenario/scenario_ubuntu2204-custom-ca-trust.go:33]: From what I understand DirectoryValidator checks for specific file name, the way CustomCA works is it generates a new name for file after copying, I could technically hardcode that name there, but all I really care about in this test is that one file exists in this dir.

**Review by cameronmeissner — APPROVED** [review]

File: e2e/scenario/validators.go [hunk 2]
```diff
@@ -65,3 +65,16 @@ func NvidiaSMINotInstalledValidator() *LiveVMValidator {
...
+
+func NonEmptyDirectoryValidator(dirName string) *LiveVMValidator {
+	return &LiveVMValidator{
+		Description: "assert that there are files in directory",
+		Command:     fmt.Sprintf("ls -1q %s | grep -q '^.*$' && true || false", dirName),
+		Asserter: func(code, stdout, stderr string) error {
+			if code != "0" {
+				return fmt.Errorf("expected to find a file in directory %s, but did not", dirName)
+			}
+			return nil
```
Comment (cameronmeissner) [inline e2e/scenario/validators.go:77]: what happened 🍡

Comment (cameronmeissner) [reply e2e/scenario/validators.go:77]: nvm nvm nvm

**Pull request merged**